### PR TITLE
Utt user storage

### PR DESCRIPTION
**Pull request by yontyon**
* **Problem Overview**  
  Currently, a utt user is unrecoverable due to a lack of persistent storage. In this PR we introduce a minimal storage API for the utt user that allows the user to recover. In addition, we introduce a basic file-based persistent storage implementation
* **Testing Done**  
  - unitests for FileBasedUserStorage component
  - e2e tests

see #2883

### CODE REVIEW

**Review by teoparvanov — APPROVED** [review]
Looks great, thanks @yontyon !

File: utt/CMakeLists.txt [hunk 1]
```diff
@@ -38,13 +38,13 @@ endif()
...
   "Enable building of wallet-cli"
-  OFF
+  ON
 )
 
 option(
   BUILD_PRIVACY_ADMIN_CLI
   "Enable building of admin-cli"
-  OFF
+  ON
```
Comment (teoparvanov) [inline utt/CMakeLists.txt:47]: @yontyon, is this change intentional?

Comment (yontyon) [reply utt/CMakeLists.txt:47]: Yes, otherwise the storage tests won't run as part of the CI

Comment (teoparvanov) [reply utt/CMakeLists.txt:47]: Thanks for the clarification, makes sense!
